pegged/peg.d

```diff
@@ -31,19 +31,19 @@ import std.typetuple;
 package string stringified(scope string inp) pure
 {
     import std.format : format;
+    import std.range : only;
 
     if (inp is null)
         return `"end of input"`;
 
-    string[1] shell = [inp];
 	/* TODO: replace `format` with call to substitute!(isASCIIControlChar) that
 	 * only does escaping of control characters to make this `nothrow`. Reuse
 	 * http://mir-algorithm.libmir.org/mir_format.html#.printEscaped. */
 
     // FIXME: referencing local variable on non-scope parameter requires .idup
     // to be @safe. Compiler is wrong about scope inference:
     // https://issues.dlang.org/show_bug.cgi?id=20674
-    return format!"%(%s%)"((() @trusted => shell[].idup)());
+    return inp.idup.only.format!"%(%s%)";
```

Comment:
I think the fact that this works is still indirectly related about the same issue: https://github.com/dlang/dmd/pull/14236 . Because `only` uses a nested function internally, although I think this is way cleaner than before and the problem is deferred to phobos.

Reply:
yes, most likely